### ISSUE TITLE
Certain tests are never run, they are marked as "Content" in FSharp.Core.UnitTests

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#9515]
While working on some issue, I was searching for existing tests for `string` which landed me at `OperatorsModule2.fs`. However, searching for the [actual test `[<Test>] test.string()` ](https://github.com/dotnet/fsharp/blob/37d0cccec23612e9a86290b193aa6c86178f2d78/tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModule2.fs#L518) inside the Test Window turned up nothing.

I also noticed that the coloring was way off, and type inference did nothing. For instance notice the "unused var" color on `result`:

![image](https://user-images.githubusercontent.com/16015770/85206611-9ac43a80-b323-11ea-9675-3e502a4bfeb0.png)

After some :exploding_head:  it dawned to me and I found this:

![image](https://user-images.githubusercontent.com/16015770/85206556-5042be00-b323-11ea-97cb-4eb520fbc020.png)

In other words, it is marked as `Content`. Is this deliberate? It is also true for some, but not all, other files in `FSharp.Core.UnitTests`, which suggests that the tests are not run. Can I change it so that it actually compiles, or should I only ever do that local-only?

EDIT: it probably is deliberate, as there's a bunch of code there that doesn't compile when I enable it. I wanted to work on https://github.com/dotnet/fsharp/issues/7958 and possibly this https://github.com/dotnet/fsharp/issues/9153, but that would mean re-enabling these tests, or putting them elsewhere.

@vzarytovskii, I believe you are working on restructuring the tests, what's the best cause of action here?

**Comment by abelbraaksma** [comment]
Also, it kills VS if I re-enable that file.

**Comment by cartermp** [comment]
@abelbraaksma it's concerning that this would tank VS by enable it as an item include. We have far, far larger tests in that project that work fine (if a bit sluggish since one of them is over 50k lines of code...).

**Comment by vzarytovskii** [comment]
@abelbraaksma I would put new tests for string in a separate module(s) inside the unit tests project (maybe under String/), under appropriate subcategories (e.g. String/Equality.fs, String/Operators.fs, String/Conversion.fs, etc).

As for existing/disabled tests - I have plans to review them and move around to appropriate modules.

**Comment by abelbraaksma** [comment]
@cartermp, while making another coffee (waiting for the VS to come back), VS crashed + restarted. The cause is possibly similar to https://github.com/dotnet/fsharp/issues/9201, because there were errors and this solution also uses a lot of SRTP. 

Reason for many errors is:

```f#
open SystematicUnitTests.LibraryTestFx
```

which doesn't exist and perhaps used to define stuff like `CheckThrowsOverflowException`.

As these crashes go, after the restart, repeating the same actions _did not_ lead to another crash and just showed the errors. Odd...

**Comment by abelbraaksma** [comment]
> I would put new tests for string in a separate module(s) inside the unit tests project (maybe under String/), under appropriate subcategories (e.g. String/Equality.fs, String/Operators.fs, String/Conversion.fs, etc).

@vzarytovskii  Hmm, ok, but that sounds like the long-term plan. Perhaps for the time being it's best if I'll just re-enable, fix the compile errors, and add tests to this file so that later they can be moved to their resp. new locations. 

I don't see any other location where `string: 'a -> string` is being tested (in fact, there's virtually nothing), so for my up-and-coming PR I propose to just add a bunch here, as opposed to trying and creating a full new structure.

**Comment by cartermp** [comment]
@abelbraaksma that seems reasonable

**Comment by vzarytovskii** [comment]
> > I would put new tests for string in a separate module(s) inside the unit tests project (maybe under String/), under appropriate subcategories (e.g. String/Equality.fs, String/Operators.fs, String/Conversion.fs, etc).
> 
> 
> 
> @vzarytovskii  Hmm, ok, but that sounds like the long-term plan. Perhaps for the time being it's best if I'll just re-enable, fix the compile errors, and add tests to this file so that later they can be moved to their resp. new locations. 

Yeah, this is more of a long-term plan for existing  and new tests.

For the tests you mentioned, it’s perfectly fine to just re-enable and fix them.

> 
> I don't see any other location where `string: 'a -> string` is being tested (in fact, there's virtually nothing), so for my up-and-coming PR I propose to just add a bunch here, as opposed to trying and creating a full new structure. 
 
Sounds good to me!

**Comment by abelbraaksma** [comment]
World of oddities today. After I changed the property from `Content` to `F# Compiler`, it moved up the tree in Solution Explorer.

Notice that `LibraryTestFx.fs` is defined _after_ `SeqModule.fs`, but the latter actually uses the former. I don't know why I don't see the right order in the Solution Explorer, I thought that F# was always compilation-order dependent.

![image](https://user-images.githubusercontent.com/16015770/85207385-bbdb5a00-b328-11ea-9735-32eddad05da2.png)

Looking at the fsproj file, the order is correct (except for the strange "putting on top" of `OperatorModule2.fs`). Which begs the question why Solution Explorer doesn't show it in the right order:

```xml
  <ItemGroup>
    <Compile Include="FSharp.Core\OperatorsModule2.fs" />
    <Compile Include="NUnitFrameworkShims.fs" />
    <Compile Include="LibraryTestFx.fs" />   <!-- offending file is on top here, just not in Sln Expl -->

    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\Utils.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ArrayModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ArrayModule2.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\Array2Module.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\Array3Module.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\Array4Module.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ArrayProperties.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ComparisonIdentityModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\HashIdentityModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ListModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ListModule2.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ObsoleteListFunctions.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ListType.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\ListProperties.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\MapModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\MapType.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\SetModule.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\SetType.fs" />
    <Compile Include="FSharp.Core\Microsoft.FSharp.Collections\SeqModule.fs" />
```

**Comment by abelbraaksma** [comment]
I think I got it. So there was this:

```xml
  <ItemGroup>
    <Content Include="**/*" Exclude="**/*.bak;Directory.Build.Props;Directory.Build.targets;FSharp.Core.UnitTests.fsproj" CopyToOutputDirectory="never" />
  </ItemGroup>
```

Which showed these now-disabled `*.fs` files. After re-enabling one, VS apparently had not idea where to put it. I'll just manually edit it, I'm not too surprised that the VS tooling doesn't know what to do with this.

**Comment by abelbraaksma** [comment]
This is now fixed in the PR.

**Comment by ThorstenReichert** [comment]
> I don't see any other location where `string: 'a -> string` is being tested (in fact, there's virtually nothing), so for my up-and-coming PR I propose to just add a bunch here, as opposed to trying and creating a full new structure.

@abelbraaksma, some tests for `string` (with numerical arguments) can be found in 
tests/fsharp/Compiler/Libraries/Core/LanguagePrimitives/StringFormatTests.fs, see e.g. #9435.

**Comment by abelbraaksma** [comment]
@ThorstenReichert, great! I needed that, that'll save me some time :)
